src/levanter/data/text.py

```diff
@@ -580,6 +657,44 @@ def get_shard_source(self, split) -> ShardedDataSource[str]:
         return TextDataSource(self, split)
 
 
+@dataclass
+class LMMixtureDatasetConfig:
+    """This class represents a mixture of datasets with their associated weights.
+    It provides a consistent interface as LMDatasetConfig.
+    """
+
+    datasets: List[LMDatasetConfig]
+
+    def __post_init__(self):
+        if len(self.datasets) == 0:
+            raise ValueError("Must provide at least one dataset")
+
+        self.weights = [d.weight for d in self.datasets]
+
+        # ensure all datasets use the same tokenizer
+        if len(set([d.tokenizer for d in self.datasets])) != 1:
```

Comment:
maybe we shouldn't use LMDatasetConfig then.

What about LMDatasetSourceConfig that is just (splits of) URLs or a hf dataset it, then LMDatasetConfig takes a list of those (or map) along with weights as well as a tokenizer?

I'm not super concerned about making sure the configs are backwards compatible. Our user base is low enough ATM that we can break without making too many people mad, so I'd rather get the right thing while we still can